modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -78,199 +31,35 @@
     implements ObjectTypeParametersResolver {
 
   protected final MuleContext muleContext;
-  private Map<String, Object> parameters = new HashMap<>();
+  protected Map<String, Object> parameters = new HashMap<>();
+  protected ParametersResolverSetResolver parametersResolver;
```

Comment:
It does quite a lot more than that. It also checks exclusive optionals, etc... My point was that it has the Resolver word twice and thus it's hard to read, pronounce and understand